src/linear_algebra/special_linear_group.lean

```diff
@@ -96,75 +90,70 @@ section coe_lemmas
+  (A * B).to_lin' = B.to_lin'.trans A.to_lin' :=
+linear_equiv.to_linear_map_injective $ matrix.to_lin'_mul A B
 
-end coe_lemmas
+lemma to_lin'_apply (A : special_linear_group n R) (v : n → R) :
+  A.to_lin' v = matrix.to_lin' ↑ₘA v := rfl
 
-instance group : group (special_linear_group n R) :=
-{ mul_assoc := λ A B C, by { ext, simp [matrix.mul_assoc] },
-  one_mul := λ A, by { ext, simp },
-  mul_one := λ A, by { ext, simp },
-  mul_left_inv := λ A, by { ext, simp [adjugate_mul] },
-  ..special_linear_group.has_mul,
-  ..special_linear_group.has_one,
-  ..special_linear_group.has_inv }
+lemma to_lin'_symm_apply (A : special_linear_group n R) (v : n → R) :
+  A.to_lin'.symm v = matrix.to_lin' ↑ₘ(A⁻¹) v := rfl
 
-/-- `to_linear_equiv A` is matrix multiplication of vectors by `A`, as a linear equivalence. -/
-def to_linear_equiv (A : special_linear_group n R) : (n → R) ≃ₗ[R] (n → R) :=
-{ inv_fun := A⁻¹.to_lin',
-  left_inv := λ x, calc
-    A⁻¹.to_lin'.comp A.to_lin' x
-        = (A⁻¹ * A).to_lin' x : by rw [←to_lin'_mul]
-    ... = x : by rw [mul_left_inv, to_lin'_one, id_apply],
-  right_inv := λ x, calc
-    A.to_lin'.comp A⁻¹.to_lin' x
-        = (A * A⁻¹).to_lin' x : by rw [←to_lin'_mul]
-    ... = x : by rw [mul_right_inv, to_lin'_one, id_apply],
-  ..matrix.to_lin' A }
+/-- `matrix.special_linear_group.to_lin'` as a `monoid_hom`. -/
+@[simps]
+def to_lin'_hom : special_linear_group n R →* (n → R) ≃ₗ[R] (n → R) :=
+{ to_fun := to_lin', map_one' := to_lin'_one, map_mul' := to_lin'_mul }
 
 /-- `to_GL` is the map from the special linear group to the general linear group -/
-def to_GL (A : special_linear_group n R) : general_linear_group R (n → R) :=
-general_linear_group.of_linear_equiv (to_linear_equiv A)
+def to_GL : special_linear_group n R →* general_linear_group R (n → R) :=
+(general_linear_group.general_linear_equiv _ _).symm.to_monoid_hom.comp to_lin'_hom
 
-lemma coe_to_GL (A : special_linear_group n R) :
-  (@coe (units _) _ _ (to_GL A)) = A.to_lin' :=
-rfl
+lemma coe_to_GL (A : special_linear_group n R) : ↑A.to_GL = A.to_lin'.to_linear_map := rfl
 
-@[simp]
-lemma to_GL_one : to_GL (1 : special_linear_group n R) = 1 :=
-by { ext v i, rw [coe_to_GL, to_lin'_one], refl }
+-- this section should be last to ensure we do not use it in lemmas
+section coe_fn_instance
+
+/-- This instance is here for convenience, but is not the simp-normal form. -/
+instance : has_coe_to_fun (special_linear_group n R) :=
+{ F   := λ _, n → n → R,
+  coe := λ A, A.val }
 
 @[simp]
-lemma to_GL_mul (A B : special_linear_group n R) :
-  to_GL (A * B) = to_GL A * to_GL B :=
-by { ext v i, rw [coe_to_GL, to_lin'_mul], refl }
-
-/-- `special_linear_group.embedding_GL` is the embedding from `special_linear_group n R`
-  to `general_linear_group n R`. -/
-def embedding_GL : (special_linear_group n R) →* (general_linear_group R (n → R)) :=
-⟨λ A, to_GL A, by simp, by simp⟩
+lemma coe_fn_eq_coe (s : special_linear_group n R) : ⇑s = (s : matrix n n R) := rfl
```

Comment:
```suggestion
lemma coe_fn_eq_coe (s : special_linear_group n R) : ⇑s = (↑ₘs : matrix n n R) := rfl
```
If one doesn't provide explicitly the desired cast on the RHS, the lemma is a tautology.

Reply:
Nice catch!